NumRush.py

```diff
@@ -347,7 +347,7 @@ def run(self):
                         if(boxx > gap):
                             boxx -= gap
                     elif (event.key == K_RIGHT or event.key == K_d):
-                        if(boxx < ResX - gap):
+                        if(boxx < (ResX - (2 * gap) + 20)):
```

Comment:
How'd you get the figures?

Reply:
In the previous condition, it checks if the x-axis position of box is lesser than the difference of entire width and the gap. This condition also allows to make a right key move even if `ResX - gap` is just lesser than `boxx`. So in some dimensions it would allow for the right key move even when there isn't enough width left to fit in the gap and the box dimension causing the disappearance of the box. Now, the condition checks if there's enough room to fit the dimension of a gap, box and some extra space after the gap for the right key move to be successful.

Reply:
Sounds good, thanks!